### ISSUE TITLE
Regression output with LSTM input layer

### ISSUE THREAD

**Issue by Pashton** [fchollet/keras#667]
Hi,

I was wondering if there is an example for an lstm layer that eventually feeds into a regression output layer. I am having a hard time putting one together.

I am hoping to feed in one sample at a time, where sample looks like this: 

input: [[x,x,x,x],[x,x,x,x],[[x,x,x,x],] output: [y]

This way, I would expect an lstm layer that takes in 4 inputs and after 3 iterations produces an output h to feed into a regression layer.

Cheers

**Comment by sergeyf** [comment]
I have an example similar to yours but the raw input is actually more like: 

`[0,0,0,45,98,43,23]`

The integers are indices which are fed into an embedding layer, and 0 is a masking index. If that is helpful, let me know and I'll post some code.

**Comment by Pashton** [comment]
I don't necessarily need an embedding layer. I would just to have an lstm take in a sequence of vectors as input.

If you feel it's still related I would appreciate you sharing the code.

**Comment by sergeyf** [comment]
In that case it is probably not sufficiently related, sorry.

**Comment by fchollet** [comment]
`input: [[x,x,x,x],[x,x,x,x],[[x,x,x,x],] output: [y]`

How is this not the canonical use case of LSTM in Keras? What problem are you having?

**Comment by Pashton** [comment]
I got it running. I had an odd issue with theano complaining about too many parameters. I thought it might be related to my code, it looks like it wasn't.

I will leave my implementation here in case others are looking for the same thing.

def gen_data() :
    X, Y = [], []
    for x in xrange(1000) :
        X.append(np.random.random_sample(5,))
        Y.append(np.random.random_sample())
    return np.asarray(X), np.asarray(Y)
def temp_data() :
    X, Y = [], []
    for x in xrange(1000) :
        sample = []
        for x in xrange(4) :
            sample.append(np.random.random_sample(5,))
        X.append(sample)
        Y.append(np.random.random_sample())
    return np.asarray(X), np.asarray(Y)
if __name__ == '__main__':
	X,Y =  temp_data()
	model = Sequential()
	model.add(LSTM(5, 10, activation='sigmoid', inner_activation='hard_sigmoid'))
	model.add(Dropout(0.5))
	model.add(Dense(10,1, init='uniform', activation='linear'))
	sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
	model.compile(loss='mean_squared_error', optimizer=sgd)

	model.fit(X,Y, nb_epoch=20, batch_size=10)
	print model.evaluate(X,Y, batch_size=10)
